client/data/marketplace/use-site-search-es-query.ts

```diff
@@ -0,0 +1,217 @@
+import phpUnserialize from 'phpunserialize';
+import { UseQueryResult, UseQueryOptions, InfiniteData, useInfiniteQuery } from 'react-query';
+import { useSelector } from 'react-redux';
+import { extractSearchInformation, normalizePluginsList } from 'calypso/lib/plugins/utils';
+import { getCurrentUserLocale } from 'calypso/state/current-user/selectors';
+import {
+	DEFAULT_FIRST_PAGE,
+	DEFAULT_PAGE_SIZE,
+	DEFAULT_CATEGORY,
+	BASE_STALE_TIME,
+	SITE_SEARCH_CACHE_KEY,
+	SITE_SEARCH_ENDPOINT,
+} from './constants';
+import { ESHits, ESResponse, Plugin, PluginQueryOptions } from './types';
+import { getPluginsListKey } from './utils';
+
+/**
+ *
+ * @param pluginSlug
+ * @param iconsArray
+ * @returns An object containing a resolution -> icon url map
+ */
+const createIconsObject = ( pluginSlug: string, iconsArray: string ) => {
+	type Icon = {
+		resolution: string;
+		filename: string;
+		location: string;
+		revision: string;
+	};
+	const icons: Record< string, Icon > = phpUnserialize( iconsArray );
+	return Object.values( icons ).reduce(
+		( prev: Record< string, string >, { resolution, filename, location, revision } ) => {
+			prev[
+				resolution
+			] = `https://ps.w.org/${ pluginSlug }/${ location }/${ filename }?rev=${ revision }`;
+			return prev;
+		},
```

Comment:
Will we always have those 4 parameters? (resolution, filename, location, revision)

Do you know if there is any kind of fallback for the times we don't have some of them?

Reply:
The icons array contains these keys after converted to JSON: 

```json
{
  "icon-128x128.png": 
    "filename": "icon-128x128.png"
    "locale": ""
    "location": "assets"
    "resolution": "128x128"
    "revision": 2366418
}
```

So far, all results contained those 4 keys, and I couldn't find any results where the `locale` key is used. As this is originally stored as a PHP array, I wasn't able to filter for it in ES.

I don't know what could be a sensible fallback for these, as the index doesn't contain anything else regarding icons. For now, the `reduce` returns an empty array if there are no icons in the `assets` folder, and the default icon is used on the UI.

Reply:
At first, I was thinking about two possible fallbacks:
* resolution: but we should be handling it later down the line when we choose which version to pick
* revision: I was wondering it was sequential like `1, 2, 3, etc`, so we could use a fallback to `1`. Thanks for adding a response example here as I was not guessing it right.

That said, I think the current version is fine. Ty.

Reply:
For the plugins that have icon assets uploaded to WPORG, these fields are always indexed. We have to have a fallback for plugins that don't have icons uploaded, I've just found out that WPORG generates a pattern for these (eg. [here](https://wordpress.org/plugins/slideshare/)). I'll incorporate a similar fallback in a follow-up PR.